src/main/java/workers/Scheduler.java

```diff
@@ -103,6 +103,17 @@ public Schedule createBasicSchedule(ArrayList<Course> courses) {
         List<Section> courseLectures = c.getLectures();
         List<Section> courseTutorials = c.getTutorials();
 
+        if (courseTutorials.size() == 0){
```

Comment:
we would need to add respective changes in populatePermutations and extendPermutations for this to affect permutationScheduler.

Reply:
ah yeaj good catch

Reply:
I just need to change extendPermutations right? then both populatPermutations & extendPermutations are altered and therefore permutationScheduler has the bug fixed